File: php/tests/AddBorderTest.php
<?php

use PHPUnit\Framework\TestCase;
require "src/add_border.php";

final class AddBorderTest extends TestCase
{
    public function testAddBorder()
    {
        $data = ["abc", "ded"];
        $actual = add_border($data);
        $expected = [
            "*****",
            "*abc*",
            "*ded*",
            "*****"
        ];
        $this->assertEquals($expected, $actual);
  }
}

File: go/main/add_border_test.go
package main

import (
	"reflect"
	"testing"
)

func TestAddBorder(t *testing.T) {
	data := []string{"abc", "ded"}
	got := AddBorder(data)
	expected := []string{"*****", "*abc*", "*ded*", "*****"}
	if !reflect.DeepEqual(got, expected) {
		t.Errorf("AddBorder(%s) = %s; want %s", data, got, expected)
	}
}

